### PR TITLE
Restaurants/Cuisine models and controllers

### PR DESCRIPTION
**Pull request by hemdittman**

### CODE REVIEW

File: app/controllers/cuisines_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if @cuisine.destroy
+      head :ok
+    else
+      render json: @cuisine.errors.messages
+    end
+  end
+
+  private
+
+  def rest_params
```
Comment (yonidor) [inline app/controllers/cuisines_controller.rb]: copy paste :)

**Review by yonidor — APPROVED** [review]
image

File: app/controllers/cuisines_controller.rb (hunk 1 above)
Comment (hemdittman) [reply app/controllers/cuisines_controller.rb]: haha :sweat_smile:

File: app/controllers/cuisines_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  private
+
+  def cuisine_params
+    params.permit(:name)
+  end
+
+  def query_cuisine
+    @cuisine = Cuisine.where(id: params[:id]).first
```
Comment (ronits-wework) [inline app/controllers/cuisines_controller.rb]: you can use Cuisine.find

File: app/controllers/cuisines_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+class CuisinesController < ApplicationController
+  before_action :query_cuisine, only: [:show, :update, :destroy]
+
+  def index
+    @cuisines = Cuisine.all
```
Comment (shem8) [inline app/controllers/cuisines_controller.rb]: Why do you need it as a member (`@`)?

File: app/controllers/cuisines_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    render json: @cuisines
+  end
+
+  def show
+    if @cuisine
+      render json: @cuisine
+    else
+      render json: {error: "not-found"}, status: 404
```
Comment (shem8) [inline app/controllers/cuisines_controller.rb]: Use strings file

Comment (shem8) [inline app/controllers/cuisines_controller.rb]: `:not_found`

File: app/controllers/cuisines_controller.rb (hunk 2 above)
Comment (shem8) [inline app/controllers/cuisines_controller.rb]: `params.require`

Comment (shem8) [reply app/controllers/cuisines_controller.rb]: or even better: `find!`

File: app/models/cuisine.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class Cuisine < ApplicationRecord
```
Comment (shem8) [inline app/models/cuisine.rb:11]: pls annotate your models

File: db/migrate/20180321101111_create_cuisines.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CreateCuisines < ActiveRecord::Migration[5.1]
+  def change
+    create_table :cuisines do |t|
+      t.string :name, null: false
+
```
Comment (shem8) [inline db/migrate/20180321101111_create_cuisines.rb:5]: so many spaces =0

File: app/controllers/cuisines_controller.rb (hunk 4 above)
Comment (hemdittman) [reply app/controllers/cuisines_controller.rb]: :+1:

File: db/migrate/20180321101111_create_cuisines.rb (hunk 6 above)
Comment (hemdittman) [reply db/migrate/20180321101111_create_cuisines.rb:5]: :+1:

File: app/controllers/cuisines_controller.rb (hunk 2 above)
Comment (hemdittman) [reply app/controllers/cuisines_controller.rb]: :+1:

File: app/controllers/cuisines_controller.rb (hunk 3 above)
Comment (hemdittman) [reply app/controllers/cuisines_controller.rb]: removing.

**Review by GeReV — APPROVED** [review]
LOKTM

**Review by shem8 — APPROVED** [review]

**Pull request merged**